### ISSUE TITLE
Switch xrwrap into the pattern of an XArray Accessor

### ISSUE THREAD

**Issue by pbranson** [iosonobert/zutils#1]
There is some useful stuff here, but it would be nice to use the Accessor pattern, which xrwarp essentially does but not quite.

That would allow you to open a dataset with 'native' xarray and access your function namespace like this:

```
import xarray as xr
ds=xr.open_dataset('ncfile.nc')
ds.xrwrap.add_history('isonobert was here')
```
without have to call wrap(ds) https://github.com/iosonobert/zutils/blob/master/zutils/xrwrap.py#L150

**Comment by pbranson** [comment]
See http://xarray.pydata.org/en/stable/internals/extending-xarray.html?highlight=accessor#extending-xarray

**Comment by iosonobert** [comment]
Yeah that's way better - didn't know about accessors. I'd be very keen to implement this. I just added you to the pIMOS repo we discussed ages ago. I haven't worked on it much of late but it's a bit of an indicator as to how we intended to use this xrwrap. Each instrument subclasses it and adds a few methods. I see that with the xarray accessors you need to have only self and ds as inputs to __init__. This isn't the way I had set it up, so would need to do some interface changes before implementing this. Thanks for the pointer - hopefully I'll get time to get into this this before the end of the year.
